pkg/storage/cloud/s3_storage.go

```diff
@@ -55,14 +56,20 @@ func s3QueryParams(conf *roachpb.ExternalStorage_S3) string {
 }
 
 func makeS3Storage(
-	ctx context.Context, conf *roachpb.ExternalStorage_S3, settings *cluster.Settings,
+	ctx context.Context,
+	ioConf base.ExternalIOConfig,
+	conf *roachpb.ExternalStorage_S3,
+	settings *cluster.Settings,
 ) (ExternalStorage, error) {
 	if conf == nil {
 		return nil, errors.Errorf("s3 upload requested but info missing")
 	}
 	region := conf.Region
 	config := conf.Keys()
 	if conf.Endpoint != "" {
+		if ioConf.DisableHTTP {
+			return nil, errors.New("Custom endpoints disallowed for s3")
```

Comment:
nit: imo this error would be a bit more helpful if it explained why -- that the http disallow flag was the reason.

Reply:
additionally, error messages do not start with a capital - they are not sentences and can be embedded in larger sentences

Reply:
Done.